widget.py

```diff
@@ -0,0 +1,309 @@
+"""
+Module for centering switches and handling mouse events.
+"""
+import itertools
+from matplotlib.widgets import Button, SpanSelector, Slider, RadioButtons
+import crystlib
+
+
+class Bttn_bins(Button):
+    """
+    Klasa reprezentująca guzik do zmiany ilości
+    bins we wszystkich histogrmaach
+    """
+    __bins = None
+    # zmienna używana przez wszystkie objekty tej klasy
+    # inicjalizowana przez set_bins
+
+    def __init__(self, fig, histogram_list, ax, label):
+        # Initialize parent constructor.
+        super(Bttn_bins, self).__init__(ax=ax, label=label)
+        self.fig = fig  # For calling canvas draw which refreshes the image.
+        self.histogram_list = histogram_list  # List with all histograms
+        self.label = label
+        super(Bttn_bins, self).on_clicked(self.change_bins)  # Click reaction
+
+    def set_bins(bins):
+        """
+        Ustawia mi początkową wartość bins dla wszystkich
+        histogrmaów
+        """
+        Bttn_bins.__bins = bins
+
+    def change_bins(self, event):
+        """
+        Method for keyboard handling.
+        """
+        # Changing number of bins; Number is a power of 2. Max val. 512.
+
+        if self.label == '+':
+            if Bttn_bins.__bins < 512:
+                Bttn_bins.__bins *= 2
+                for his in self.histogram_list:
+                    his.set_bins(Bttn_bins.__bins)
+                    his.update()
+                    his.draw_green_space()
+        elif self.label == '-':
+            if Bttn_bins.__bins > 2:
+                Bttn_bins.__bins /= 2
+                for his in self.histogram_list:
+                    his.set_bins(Bttn_bins.__bins)
+                    his.update()
+                    his.draw_green_space()
+        self.fig.canvas.draw()
+
+
+class Radio(RadioButtons):
+    """
+    class used in hdfsee
+    image - image created by functions imshow()
+    fig - object of the Figuge class used to redraw
+    cmap - color map used to change color map in image
+    """
+    def __init__(self, fig, ax, labels, cmap, image):
+        self.image = image
+        self.fig = fig
+        self.cmap = cmap
+        # Initialize parent constructor.
+        super(Radio, self).__init__(ax=ax, labels=labels, active=0)
+        # On click reaction.
+        self.on_clicked(self.colorfunc)
+
+    def colorfunc(self, label):
+        """
+        Calls itself after switching colour.
+        Updates the image.
+        """
+        # change name cmap
+        self.cmap = label
+        # set cmap
+        self.image.set_cmap(cmap=self.cmap)
+        # Redraw the current figure.
+        self.fig.canvas.draw()
+
+    def set_image(self, image):
+        """
+        set a new object returned by imshow
+        """
+        self.image = image
+
+    def get_cmap(self):
+        """
+        returned last cmap
+        """
+        return self.cmap
+
+
+class My_slider(Slider):
```

Comment:
Do not use names like `My_slider`. This kind of naming is used for examples but it does not give useful information about what this does. What is the purpose of this class?

Reply:
Addressed in: e73a1996e75cfceb523bf223bbad6228e6e19c81